deps/v8/src/libplatform/default-platform.cc

```diff
@@ -34,6 +34,18 @@ const int DefaultPlatform::kMaxThreadPoolSize = 8;
 DefaultPlatform::DefaultPlatform()
     : initialized_(false), thread_pool_size_(0) {}
 
+void DefaultPlatform::ForkingCleanup() {
+  base::LockGuard<base::Mutex> guard(&lock_);
+  queue_.PurgeWorkers();
+  if (initialized_) {
+    for (auto i = thread_pool_.begin(); i != thread_pool_.end(); ++i) {
+      delete *i;
```

Comment:
A key point here, which was non obvious to me, is that the WorkerThread destructor calls `Thread.Join`, and so will block until the thread is done.